tests/ratelimit.rs

```diff
@@ -56,3 +60,46 @@ fn multiple() {
         elapsed
     );
 }
+
+#[test]
+fn stream() {
+    let i = Instant::now();
+    let lim = DirectRateLimiter::<LeakyBucket>::per_second(nonzero!(10u32));
+    let mut stream = stream::repeat::<_, io::Error>(()).ratelimit(lim).wait();
+
+    for _ in 0..10 {
+        stream.next().unwrap().unwrap();
+    }
+    assert!(i.elapsed() <= Duration::from_millis(100));
+
+    stream.next().unwrap().unwrap();
+    assert!(i.elapsed() > Duration::from_millis(100));
+    assert!(i.elapsed() <= Duration::from_millis(200));
+
+    stream.next().unwrap().unwrap();
+    assert!(i.elapsed() > Duration::from_millis(200));
+    assert!(i.elapsed() <= Duration::from_millis(300));
+}
```

Comment:
This isn't something for you to address in this PR (just musing out loud!), but I feel like having these tests depend on wall-clock time is a little displeasing. I think I'll implement a testable time source in `ratelimit_meter` sometime - maybe we can build something with https://docs.rs/quanta/0.3.0/quanta/.